tests/unit_tests/huggingface_pipelines/text.py

```diff
@@ -0,0 +1,242 @@
+    assert all(isinstance(text, list) for text in result["embedding_text"])
+    assert all(
+        isinstance(item, str)
+        for sublist in result["embedding_text"]
+        for item in sublist
+    )
+
+
+@pytest.mark.parametrize(
+    "invalid_batch",
+    [
+        {"text": "Not a list"},
+        {"text": [1, 2, 3]},
+    ],
+)
+def test_text_to_embedding_invalid_input(
+    text_to_embedding_config, mock_text_to_embedding_model, invalid_batch
+):
+    pipeline = HFTextToEmbeddingPipeline(text_to_embedding_config)
+    with pytest.raises(ValueError):
+        pipeline.process_batch(invalid_batch)
+
+
+@pytest.mark.parametrize(
+    "invalid_batch",
+    [
+        {"embedding": "Not a list"},
+        {"embedding": [1, 2, 3]},
+    ],
+)
+def test_embedding_to_text_invalid_input(
+    embedding_to_text_config, mock_embedding_to_text_model, invalid_batch
+):
+    pipeline = HFEmbeddingToTextPipeline(embedding_to_text_config)
+    with pytest.raises(ValueError):
+        pipeline.process_batch(invalid_batch)
+
+
+def test_text_to_embedding_large_batch(
+    text_to_embedding_config, mock_text_to_embedding_model
+):
+    pipeline = HFTextToEmbeddingPipeline(text_to_embedding_config)
+    large_batch = {"text": [["Hello"] * 100, ["World"] * 100]}
+    result = pipeline.process_batch(large_batch)
+    assert "text_embedding" in result
+    assert len(result["text_embedding"]) == 2
+    assert all(len(emb) == 100 for emb in result["text_embedding"])
+
+
+def test_embedding_to_text_large_batch(
+    embedding_to_text_config, mock_embedding_to_text_model
+):
+    pipeline = HFEmbeddingToTextPipeline(embedding_to_text_config)
+    large_batch = {
+        "embedding": [
+            [np.array([0.1, 0.2, 0.3])] * 100,
+            [np.array([0.4, 0.5, 0.6])] * 100,
+        ]
+    }
+    result = pipeline.process_batch(large_batch)
+    assert "embedding_text" in result
+    assert len(result["embedding_text"]) == 2
+    assert all(len(text) == 100 for text in result["embedding_text"])
+
+
+def test_text_to_embedding_process_batch_complicated(
+    text_to_embedding_config, mock_text_to_embedding_model
+):
+    pipeline = HFTextToEmbeddingPipeline(text_to_embedding_config)
+    batch = {
+        "text": [
+            ["The quick brown fox jumps over the lazy dog."],
+            ["This is a test sentence.", "And another one."],
+            [],
+            [
+                "Short.",
+                "Medium length sentence.",
+                "Longer sentence with more words.",
+                "Very long sentence that goes on and on with many more words than the others.",
+            ],
+        ]
+    }
+    result = pipeline.process_batch(batch)
+
+    assert "text_embedding" in result
+    assert len(result["text_embedding"]) == 4
+    assert all(
+        isinstance(item, np.ndarray)
+        for sublist in result["text_embedding"]
+        for item in sublist
+    )
+    assert len(result["text_embedding"][0]) == 1
```

Comment:
Decode output tensors, to check correctness, test dtype on cuda and cpu device on notebook